src/test/java/io/cdap/plugin/sfmc/source/MarketingCloudRecordReaderTest.java

```diff
@@ -227,38 +197,154 @@ public void testFetchDataWithDataExtension() throws Exception {
+    conf.set("soapEndpoint", SOAP_ENDPOINT);
+    conf.set("useOAuth2Authentication", "true");
+    Schema schema = Schema.recordOf("record",
+                                    Schema.Field.of("id", Schema.of(Schema.Type.LONG)),
+                                    Schema.Field.of("timestamp",
+                                                    Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS))));
+    List<MarketingCloudColumn> columns = new ArrayList<>();
+    MarketingCloudColumn column = new MarketingCloudColumn("price", "String");
+    columns.add(column);
+    MarketingCloudObjectInfo sObjectInfo = Mockito.spy(new MarketingCloudObjectInfo(object, columns));
+    ETClient etClient = PowerMockito.mock(ETClient.class);
+    PowerMockito.whenNew(ETClient.class).withArguments(Mockito.anyString()).thenReturn(etClient);
+    PowerMockito.spy(new MarketingCloudClient(new ETClient(conf)));
+    MarketingCloudClient client = PowerMockito.mock(MarketingCloudClient.class);
+    PowerMockito.whenNew(MarketingCloudClient.class).withArguments(Mockito.any()).thenReturn(client);
+    PowerMockito.mockStatic(ClassLoader.class);
+    Mockito.when(client.fetchObjectSchema(object)).thenReturn(sObjectInfo);
+    Mockito.when(sObjectInfo.getSchema()).thenReturn(schema);
+    Mockito.when(client.fetchObjectRecords(object, null)).thenReturn(etResponse);
+    Mockito.when(responses.getObjects()).thenReturn((List<ETDataExtensionRow>) results);
+    marketingCloudRecordReader.initialize(split, null);
+    try {
+      Assert.assertFalse(marketingCloudRecordReader.nextKeyValue());
+    } catch (Exception e) {
+      Assert.assertEquals(79, e.getMessage().length());
+    }
   }
 
   @Test
-  public void testConvertToValueStringFieldType() {
-    Schema fieldSchema = Schema.of(Schema.Type.STRING);
-    Assert.assertNotNull(marketingCloudRecordReader.convertToValue("Field Name", fieldSchema,
-                                                                   new HashMap<>(1)));
+  public void testGetCurrentKey() {
+    MarketingCloudRecordReader marketingCloudRecordReader = new MarketingCloudRecordReader(marketingCloudSourceConfig);
+    Assert.assertNotNull(marketingCloudRecordReader.getCurrentKey());
+  }
+
+  @Test
+  public void testClose() {
+    MarketingCloudRecordReader marketingCloudRecordReader = new MarketingCloudRecordReader(marketingCloudSourceConfig);
+    marketingCloudRecordReader.close();
   }
 
   @Test
-  public void testGetCurrentKeyValue() throws IOException, NoSuchFieldException {
+  public void testGetCurrentKeyValue() throws NoSuchFieldException {
     ETApiObject row = new ETDataExtensionRow();
     row.setId("id");
     List<MarketingCloudColumn> columns = new ArrayList<>();
     MarketingCloudColumn column = new MarketingCloudColumn("name", "String");
     columns.add(column);
+    MarketingCloudRecordReader marketingCloudRecordReader1 = new MarketingCloudRecordReader(marketingCloudSourceConfig);
```

Comment:
nit: terminology

Please change it to something like newMarketingCloudRecordReader or marketingCloudRecordReaderCopy

Reply:
Done. Renamed it to 'cloudRecordReader'.